sdk-base/src/main/java/com/mapbox/maps/ExtensionUtils.kt

```diff
@@ -37,4 +40,59 @@ fun MapSnapshotInterface.bitmap(): Bitmap {
   val buffer: ByteBuffer = ByteBuffer.wrap(image.data)
   bitmap.copyPixelsFromBuffer(buffer)
   return bitmap
+}
+
+/**
+ * Calculates the center point of this [CoordinateBounds] by simple interpolation and returns
+ * it as a point. This is a non-geodesic calculation which is not the geographic center.
+ *
+ * @return [Point] center of this [CoordinateBounds]
+ */
+fun CoordinateBounds.getCenter(): Point {
+  val latCenter = (northeast.latitude() + southwest.latitude()) / 2.0
+  val lngCenter = (northeast.longitude() + southwest.longitude()) / 2.0
+
+  return Point.fromLngLat(lngCenter, latCenter)
+}
+
+/**
+ * Determines whether this [CoordinateBounds] contains a [Point].
+ *
+ * @param point the point which may be contained
+ * @return true, if the point is contained within the bounds
+ */
+fun CoordinateBounds.contains(point: Point) =
+  point.latitude() <= northeast.latitude() &&
+    point.latitude() >= southwest.latitude() &&
+    point.longitude() >= southwest.longitude() &&
+    point.longitude() <= northeast.longitude()
+
+/**
+ * Determines whether this [CoordinateBounds] contains another bounds.
+ *
+ * @param other the bounds which may be contained
+ * @return true, if the bounds is contained within the bounds
+ */
+fun CoordinateBounds.contains(other: CoordinateBounds) =
+  other.contains(northeast) && other.contains(southwest)
+
+/**
+ * Returns a new [CoordinateBounds] that stretches to contain both this and another [CoordinateBounds].
+ *
+ * @param bounds [CoordinateBounds] to add
+ * @return [CoordinateBounds]
+ */
+fun CoordinateBounds.union(bounds: CoordinateBounds): CoordinateBounds {
+  return CoordinateBounds(
+    // southwest
+    Point.fromLngLat(
+      min(this.southwest.longitude(), bounds.southwest.longitude()),
+      min(this.southwest.latitude(), bounds.southwest.latitude())
+    ),
+    // northeast
+    Point.fromLngLat(
+      max(this.northeast.longitude(), bounds.northeast.longitude()),
+      max(this.northeast.latitude(), bounds.northeast.latitude())
+    )
+  )
```

Comment:
can you also add tests?

Reply:
I was hoping you will not notice it 😢 
Yep, will do

Reply:
@pengdev - we don't even have `test` package for `sdk-base`.
Assuming current solution is a bit of workaround while we land better solution upstream - perhaps we could omit tests in that case?

Reply:
Since these functions involve some complex logic, I would suggest to add some unit tests to verify the corner cases. A few validation should be enough, especially around the border of the bounds and across the international date line. The support of wrapped and unwrapped coordinates should also be taken into consideration, and maybe state it clearly in the documentation.